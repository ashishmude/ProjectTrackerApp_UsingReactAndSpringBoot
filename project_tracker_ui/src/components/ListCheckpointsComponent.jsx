import React, { Component } from "react";
import ProjectCheckpointService from "../service/ProjectCheckpointService";
import CheckpointTasks from "./CheckpointTasks";
import {
  Container,
  AccordionDetails,
  AccordionSummary,
  Accordion,
  Typography,
  Chip,
} from "@material-ui/core";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { withStyles } from "@material-ui/core/styles";
import data from "../data.json";

const styles = (theme) => ({
  chip: {
    marginLeft: theme.spacing(1),
  },
  accordionSummary: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
  },
  container: {
    marginTop: theme.spacing(2),
  },
  title: {
    marginBottom: theme.spacing(2),
  },
  accordionDetails: {
    display: "flex",
    flexDirection: "column",
  },
  description: {
    marginBottom: theme.spacing(1),
  },
});

class ListCheckpointsComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      expanded: {},
    };
    this.refreshCheckpoints = this.refreshCheckpoints.bind(this);
  }
  componentDidMount() {
    this.refreshCheckpoints();
  }
  refreshCheckpoints() {
     ProjectCheckpointService.retrieveAllCheckpoints()
         .then(response => {
             console.log(response);
             this.setState({ data: response.data });
         })
    //this.setState({ data });
  }
  onExpandedChange = (expanded) => {
    this.setState({ expanded });
  };

  render() {
    const { data } = this.state;
    const { classes } = this.props;
    return (
      <>
        <Container className={classes.container}>
          <Typography variant="h6" className={classes.title}>
            Checkpoints
          </Typography>
          {data.map((checkpoint, index) => (
            <Accordion>
              <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
                aria-controls="panel1a-content"
                id="panel1a-header"
                className={classes.accordionSummary}
              >
                <Typography>{`${index + 1}. ${
                  checkpoint.checkpointName
                }`}</Typography>
                <Chip
                  size="small"
                  label={`${checkpoint.completionPercentage}% complete`}
                  color="primary"
                  className={classes.chip}
                />
              </AccordionSummary>
              <AccordionDetails className={classes.accordionDetails}>
                <Typography
                  variant="h6"
                  component="p"
                  className={classes.description}
                >
                  {checkpoint.description}
                </Typography>
                <CheckpointTasks
                  taskList={checkpoint.taskList}
                ></CheckpointTasks>
              </AccordionDetails>
            </Accordion>
          ))}
        </Container>
      </>
    );
  }
}

export default withStyles(styles)(ListCheckpointsComponent);
