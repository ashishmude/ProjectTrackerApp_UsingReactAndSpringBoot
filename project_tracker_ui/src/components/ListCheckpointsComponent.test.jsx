import React from 'react';
import { shallow } from 'enzyme';
import ListCheckpointsComponent from './ListCheckpointsComponent';


describe("ListCheckpointsComponent", () => {
  it("should render ListCheckpointsComponent", () => {
    const component = shallow(<ListCheckpointsComponent />);
    expect(component.getElements()).toMatchSnapshot();
  });
});
