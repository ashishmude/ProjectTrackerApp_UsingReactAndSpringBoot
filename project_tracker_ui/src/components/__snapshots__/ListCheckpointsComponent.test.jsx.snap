// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCheckpointsComponent should render ListCheckpointsComponent 1`] = `
Array [
  <ListCheckpointsComponent
    classes={
      Object {
        "accordionDetails": "ListCheckpointsComponent-accordionDetails-5",
        "accordionSummary": "ListCheckpointsComponent-accordionSummary-2",
        "chip": "ListCheckpointsComponent-chip-1",
        "container": "ListCheckpointsComponent-container-3",
        "description": "ListCheckpointsComponent-description-6",
        "title": "ListCheckpointsComponent-title-4",
      }
    }
  />,
]
`;
